src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp

```diff
@@ -68,6 +69,94 @@ void OVInferRequestDynamicTests::TearDown() {
     function.reset();
 }
 
+TEST_P(OVInferRequestDynamicTests, InferDynamicNetwork) {
+    std::vector<ov::Shape> vectorShapes{{1, 4, 20, 20}, {1, 4, 20, 20}, {1, 4, 10, 10}};
+    bool checkOutputValues = !strcmp(function->get_friendly_name().c_str(), "Sin");
+    const std::string tensor_name = "input_tensor";
+    std::map<std::string, ov::PartialShape> shapes;
+    shapes[tensor_name] = { ov::Dimension(1, 5), ov::Dimension(1, 5), ov::Dimension(10, 40), ov::Dimension(10, 40) };
+    OV_ASSERT_NO_THROW(function->reshape(shapes));
+    // Load ov::Model to target plugins
+    auto execNet = ie->compile_model(function, targetDevice, configuration);
+    // Create InferRequest
+    ov::runtime::InferRequest req;
+    for (auto& shape : vectorShapes) {
+        ov::runtime::Tensor inTensor = ov::test::utils::create_and_fill_tensor(element::f32, shape, 100, -50);
+        OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+        OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", inTensor));
+        OV_ASSERT_NO_THROW(req.infer());
+        OV_ASSERT_NO_THROW(req.start_async());
+        OV_ASSERT_NO_THROW(req.wait());
+        if (checkOutputValues) {
+            ov::runtime::Tensor outTensor = req.get_tensor("Sin");
+            for (int i = 0; i < inTensor.get_size(); i++) {
+                ASSERT_LT(fabs(sin(inTensor.data<float>()[i]) - outTensor.data<float>()[i]), std::numeric_limits<float>::epsilon());
+            }
+        }
+    }
+}
+
+TEST_P(OVInferRequestDynamicTests, InferDynamicNetworkSetOutputTensorBeforeInfer) {
```

Comment:
could you also add test case when we set output tensor which was created on pre-allocated memory. 
ctor: `Tensor(const element::Type type, const Shape& shape, void* host_ptr, const Strides& strides = {});`
and check that data pointers are equal